src/frontend/screens/MapScreen/index.tsx

```diff
@@ -160,32 +160,45 @@ export const MapScreen = ({
           </>
         )}
       </Mapbox.MapView>
+      <View style={styles.bottomContainer}>
+        <View style={{flex: 1, alignItems: 'center'}}>
+          <GPSPill onPress={() => navigation.navigate('GpsModal')} />
+        </View>
+
+        <TouchableOpacity
+          testID="MAIN.add-observation-btn"
+          onPress={handleAddPress}>
+          <AddButtonSVG />
+        </TouchableOpacity>
+
+        {coords && locationServicesEnabled ? (
+          <TouchableOpacity
+            style={{flex: 1, alignItems: 'center'}}
+            onPress={handleLocationPress}>
+            {following ? <LocationFollowingIcon /> : <LocationNoFollowIcon />}
+          </TouchableOpacity>
+        ) : (
+          <View style={{width: 0, height: 0, flex: 1}} />
+        )}
+      </View>
```

Comment:
This will allow you to render the 3 things without "magic numbers"
I got rid of some unnecssary things in the process:before there was a loading state for the button. I added that a long time ago before the map screen and suspense stuff. I don't think it is necessary anymore. I also just moved everything up into this screen instead of using nested components